tsdb/shard.go

```diff
@@ -1237,6 +1251,29 @@ func (a Shards) CreateIterator(ctx context.Context, measurement *influxql.Measur
 	return query.Iterators(itrs).Merge(opt)
 }
 
+func (a Shards) createSeriesIterator(ctx context.Context, opt query.IteratorOptions) (_ query.Iterator, err error) {
+	var (
+		idxs  []Index
+		sfile *SeriesFile
+	)
```

Comment:
Can you preallocate `idxs` to `make([]Index, len(a))`?

Reply:
Done 👍